packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -121,7 +133,11 @@ export class DetachedFieldIndex {
 	 * Returns undefined if no such id is known to the index.
 	 */
 	public tryGetEntry(id: Delta.DetachedNodeId): ForestRootId | undefined {
-		return tryGetFromNestedMap(this.detachedNodeToField, id.major, id.minor);
+		const innerRangeMap = this.detachedNodeRangeMap.get(id.major);
+		if (innerRangeMap !== undefined) {
+			return getFromRangeMap(innerRangeMap, id.minor, 1).value;
```

Comment:
This won't work because that `ForestRootId` value corresponds to the first `DetachedNodeId` in the range that has this `ForestRootId` value. For example, if the range stored in the RangeMap were `{ start: 10, length: 2, value: 40 }`, and the query were for a `DetachedNodeId.minor` equal to 11, then the current code would return `ForestRootId` `40` instead of `41`.

I think there's two options to solve this: either use `getFirstEntryFromRangeMap` and do the offset math, or change how we use the `RangeMap` by storing the offset between `DetachedNodeId`s and `ForestRootId`s for each range.

The second option is more work, but I think it will lead to less confusion/bugs like this one over time, so I think it's best to go for that. This means that instead of storing the range entry `{ start: 10, length: 2, value: 40 }` to represent the fact that 10 maps to 40 and 11 maps to 41, we would be storing the range entry `{ start: 10, length: 2, value: 30 }` instead.